### PR TITLE
Implements frontend for ProfileView (mentor/mentee)

### LINKED ISSUE #50: Profile - Display Mentor/Mentee Profiles
**Requirements for feature**
- [ ] Implement an API service to handle GET mentor/mentee (fill this to return dummy data until backend is up)
- [ ] Setup a model for mentor/mentee that can be used for Profile View and Edit
- [ ] Create Profile Display Scene and reuse logic using components etc whenever you see fit

Note - Don't link the "Book Session Via Calendly" button right now. This will be a separate task

**Any context for feature**
- Profile View Display is the screen users can view their own profile or a different user's profile. The View will change depending on who is viewing who's profile.  For example, when a mentee is viewing a mentor's profile they will see a"Book Session Via Calendly" button that they can use to schedule via calendly. In addition, they will not see who the other mentee's are for that mentor. On the other hand, when the mentor is viewing that same profile (self), they should see a view my calendly button + see all their mentees.

If you are confused on some field, please ping on slack and we can address it with our PM and designers. 

image

**Resources / tips for development**
- You should setup a model that can be used in our ViewModel and Model for the Profile Screens. You should be able to reuse the same model for the Edit Profile screen

### PR DESCRIPTION
**Pull request by YashRavipati1**
### Tracking Info

Resolves #50

Make sure your branch name conforms to: `<feature|staging|bugfix|...>/<username>/<3-4 word description separated by dashes>`. Otherwise, please rename your branch and create a new PR.

### Changes

What changes did you make?

Added the visual aspects for the different profile views, including the MVVM architecture

### Testing

How did you confirm your changes worked? 

Previews of the screens

### Confirmation of Change 

Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

image

image

image

image

### CODE REVIEW

**Review by wllmwu — CHANGES_REQUESTED** [review]
Good work, I've made some suggestions for refactoring this code to make it cleaner, more reusable, and easier to maintain in the future.

File: ALUM/ALUM/Views/ProfileViews/MenteeProfileMentorView.swift [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                    }
+                    .padding(.leading, 16)
+                    .padding(.bottom, 8)
+                    // swiftlint:disable:next line_length
+                    HStack{
+                        Text(viewModel.mentee.mentorshipGoal)
+                            .font(Font.custom("Metropolis-Regular", size: 17, relativeTo: .headline))
+                            .lineSpacing(5)
+                        Spacer()
+                    }
```
[the remark this answers is not in the document]
Comment (wllmwu) [reply ALUM/ALUM/Views/ProfileViews/MenteeProfileMentorView.swift:78]: Also, make sure to test this with long strings that wrap onto multiple lines.

File: ALUM/ALUM/Components/MenteeCard.swift [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    var body: some View {
+        Button {
+        } label: {
+            ZStack{
+                RoundedRectangle(cornerRadius: 12.0)
+                    .frame(width: 107, height: 136.91)
+                    .offset(y: 32)
+                    .foregroundColor(Color("ALUM Dark Blue"))
+                if isEmpty{
+                    Rectangle()                .frame(width: 107, height: 75.94)
```
[the remark this answers is not in the document]
Comment (YashRavipati1) [reply ALUM/ALUM/Components/MenteeCard.swift:40]: Most of the decimal values were straight from Figma, but I changed them to be more generic.

File: ALUM/ALUM/Services/APIService.swift [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        request.httpMethod = "POST"
+
+        do {
+            let (postedData, _) = try await URLSession.shared.upload(for: request, from: jsonData)
+            let decodedData = try JSONDecoder().decode(MenteePostData.self, from: postedData)
+            let confirmationMessage = "\(decodedData.name), you have succesfully signed up with the email \(decodedData.email)!"
+            print(confirmationMessage)
+        } catch {
+            print("POST failed.")
+        }
```
[the remark this answers is not in the document]
Comment (YashRavipati1) [reply ALUM/ALUM/Services/APIService.swift]: Handled in PR #60

File: ALUM/ALUM/Views/ProfileViews/MenteeProfileMentorView.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (YashRavipati1) [reply ALUM/ALUM/Views/ProfileViews/MenteeProfileMentorView.swift:78]: Edited to remove unnecessary HStacks. Already tested with longer strings that wrap.

File: ALUM/ALUM/Views/ProfileViews/MenteeProfileMentorView.swift [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+//  Created by Yash Ravipati on 3/9/23.
+//
+
+import SwiftUI
+import WrappingHStack
+
+struct MenteeProfileMentorView: View {
+    @StateObject private var viewModel = MenteeProfileViewmodel()
+    var body: some View {
+        GeometryReader { grr in
```
[the remark this answers is not in the document]
Comment (YashRavipati1) [reply ALUM/ALUM/Views/ProfileViews/MenteeProfileMentorView.swift:14]: There are actually a number of changes in addition to the calendly button. The buttons at the top of the view, the why're you paired section (which hasn't been added yet but is a part of the view), profile card(s), etc. are also differences between the views when a different user is viewing the profile. Due to these numerous differences I believe that separating the views would be better for ease of use/readability when connecting the navigation of the app. However, if you still believe that I should merge the views I can make that change.

File: backend/src/config.ts [hunk 5]
```diff
@@ -7,7 +7,7 @@ import dotenv from "dotenv";
 
 // load the environment variables from the .env file
 dotenv.config({
-  path: ".env",
+  path: "/Users/yashravipati/Documents/GitHub/ALUM-Mobile-Application/backend/env",
```
[the remark this answers is not in the document]
Comment (YashRavipati1) [reply backend/src/config.ts:10]: Reverted

**Review by AmanKAggarwal — CHANGES_REQUESTED** [review]
Great work with the layout! Few changes needed before we can merge this - 
1. Possibly merge the redundant views so that we don't have two different views (one for a mentor viewing vs mentee viewing)
2. Reduce reused code with the use of components
3. Try to remove the hardcoded pixel values as much as possible. Let's chat about this a bit cauz I wanna see the benefits of using the pixels as well if you are aware.  
       - You can use Spacers, add paddings around your Textboxes etc.
       - Doing these things makes it more resizable and easy to change in the future.

**Review by AmanKAggarwal — COMMENTED** [review]
Great stuff with the PR! Just couple of minor things - 
1. Delete `NavigationComponent` because it is unused it seems. Correct me if I am wrong
2. Update the colors as per the current figma

**Review by sidhantrohatgi — APPROVED** [review]

**Review by AmanKAggarwal — APPROVED** [review]
LGTM. Just the one change to delete the unused component

File: ALUM/ALUM/Components/NavigationComponent.swift [hunk 6]
```diff
@@ -0,0 +1,58 @@
+//
```
[the remark this answers is not in the document]
Comment (YashRavipati1) [reply ALUM/ALUM/Components/NavigationComponent.swift]: This component is actually used in both menteeprofileview and mentor profileview for when its not the user viewing their own profile.

**Review by AmanKAggarwal — APPROVED** [review]
still looks good

**Pull request closed without merging**